storage/system-test/acl.test.js

```diff
@@ -0,0 +1,237 @@
+
+var path = require('path');
+var Storage = require('@google-cloud/storage');
+var uuid = require('node-uuid');
+var program = require('../acl');
+
+var storage = Storage();
+var bucketName = 'nodejs-docs-samples-test-' + uuid.v4();
+var fileName = 'test.txt';
+var entity = 'allAuthenticatedUsers';
+var role = 'READER';
+var filePath = path.join(__dirname, '../resources', fileName);
+
+var expected = {
+  entity: entity,
+  role: role
+};
+
+describe('storage:acl', function () {
+  before(function (done) {
+    storage.createBucket(bucketName, function (err, bucket) {
+      if (err) {
+        return done(err);
+      }
+      bucket.upload(filePath, done);
+    });
+  });
+
+  after(function (done) {
+    storage.bucket(bucketName).deleteFiles({ force: true }, function (err) {
+      if (err) {
+        return done(err);
+      }
+      storage.bucket(bucketName).delete(done);
+    });
+  });
+
+  describe('add', function () {
+    it('should add access controls to a bucket', function (done) {
+      program.addAccessControl({
+        bucket: bucketName,
+        entity: entity,
+        role: role
+      }, function (err, aclObject) {
+        assert.ifError(err);
+        assert.deepEqual(aclObject, expected);
+        done();
+      });
+    });
+
+    it('should add "default" access controls to a bucket', function (done) {
+      program.addAccessControl({
+        bucket: bucketName,
+        entity: entity,
+        role: role,
+        default: true
+      }, function (err, aclObject) {
+        assert.ifError(err);
+        assert.deepEqual(aclObject, expected);
+        done();
+      });
+    });
+
+    it('should add access controls to a file', function (done) {
+      program.addAccessControl({
+        bucket: bucketName,
+        file: fileName,
+        entity: entity,
+        role: role
+      }, function (err, aclObject) {
+        assert.ifError(err);
+        assert.ifError(err);
+        assert.deepEqual(aclObject, expected);
+        setTimeout(done, 2000); // Make sure changes have time to take effect
+      });
+    });
+  });
+
+  describe('get', function () {
+    it('should get all access controls for a bucket', function (done) {
+      program.getAccessControl({
+        bucket: bucketName
+      }, function (err, aclObjects) {
+        assert.ifError(err);
+        assert(Array.isArray(aclObjects));
+        assert(aclObjects.length > 1);
+        var matchesExpected = aclObjects.filter(function (aclObject) {
+          return aclObject.entity === entity && aclObject.role === role;
+        });
+        assert.equal(matchesExpected.length, 1, 'Recently added aclObject should be in list');
+        done();
+      });
+    });
+
+    it('should get an entity\'s access controls for a bucket', function (done) {
+      program.getAccessControl({
+        bucket: bucketName,
+        entity: entity
+      }, function (err, aclObject) {
+        assert.ifError(err);
+        assert.deepEqual(aclObject, expected);
```

Comment:
Yeah, they definitely aren't strictly equal. `expected` is an object defined at the top of the file, `aclObject` came from the Storage API. The only other way to verify equality would be to be to recursively loop over properties, aka implement `assert.deepEqual`.